.github/workflows/ci.yml

```diff
@@ -314,7 +409,364 @@ jobs:
+          ./.github/gcs-download-artifacts.sh ${{ secrets.GCP_SA_KEY }} ${{ github.run_id }}/${{ matrix.name }} ${{ format('src/zcashd{0}', matrix.file_ext) }} ./src/
+          ./.github/gcs-download-artifacts.sh ${{ secrets.GCP_SA_KEY }} ${{ github.run_id }}/${{ matrix.name }} ${{ format('src/zcash-cli{0}', matrix.file_ext) }} ./src/
+
+      - name: Make artifact executable
+        if: runner.os != 'Windows' # (disabled to save costs)
+        run: |
+          chmod +x ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}
+          chmod +x ${{ format('./src/bench/bench_bitcoin{0}', matrix.file_ext) }}
+          chmod +x ${{ format('./src/zcashd{0}', matrix.file_ext) }}
+          chmod +x ${{ format('./src/zcash-cli{0}', matrix.file_ext) }}
+          chmod +x ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}
+          chmod +x ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}
+          chmod +x ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}
+
+      - name: Run sec-hard test
+        shell: bash
+        if: runner.os != 'Windows' # (disabled to save costs)
+        run: >
+          SRC_DIR=$(pwd)
+          HOST=${{ matrix.host }}
+          python3 ./qa/zcash/full_test_suite.py sec-hard
+
+  # Not working in Windows (disabled to save costs) and disabling in MacOS to save costs
+  rpc-depends:
+    name: RPC set up tier ${{ matrix.tier }} platform ${{ matrix.platform }}
+    needs: [setup, build]
+    runs-on: ${{ matrix.build_os }}
+    continue-on-error: ${{ matrix.tier != 1 }}
+    strategy:
+      matrix:
+        name: ${{ fromJson(needs.setup.outputs.test_names) }}
+        include: ${{ fromJson(needs.setup.outputs.test_matrix) }}
+
+    steps:
+      - name: get python depends
+        if: runner.os != 'Windows' && runner.os != 'macOS' # (disabled to save costs)
+        run: |
+          python3 -m venv ./venv
+          source ./venv/bin/activate
+          pip install simplejson zmq asyncio
+
+      - uses: actions/cache/save@v3
+        if: runner.os != 'Windows' && runner.os != 'macOS' # (disabled to save costs)
+        name: cache-venv
+        with:
+          enableCrossOsArchive: true
+          path: venv
+          key: venv-${{ matrix.name }}
+
+  # Not working in Windows (disabled to save costs) and disabling in MacOS to save costs
+  rpc-test:
+    name: RPC tests tier ${{ matrix.tier }} platform ${{ matrix.platform }} test ${{ matrix.rpc_tests }}
+    needs: [setup, build, rpc-depends]
+    runs-on: ${{ matrix.build_os }}
+    continue-on-error: ${{ matrix.tier != 1 }}
+    strategy:
+      matrix:
+        name: ${{ fromJson(needs.setup.outputs.rpc_test_names) }}
+        rpc_tests : ${{ fromJson(needs.setup.outputs.rpc_tests) }}
+        include: ${{ fromJson(needs.setup.outputs.rpc_test_matrix) }}
+
+    steps:
+      - uses: actions/checkout@v4
+        if: runner.os != 'Windows' && runner.os != 'macOS' # (disabled to save costs)
```

Comment:
Likewise, move this configuration into `rpc_test_matrix` instead of having it on every step (which is a maintenance hazard).

Reply:
Yes, it could go in the setup step. I designed it that way to avoid adding more minutes to the first step without knowing if the previous tests/builds would be successful and to avoid spending more minutes